source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,166 @@
+  establish to an individual server before making them available in the connection pool.
+  |
+  By keeping this option low, you reduce the risk temporary load spikes
+  from  overwhelming the database.
+
+- **maxTime()**: The *cumulative* time limit for processing operations on
+  a MongoDB server.
+  |
+  Terminating long-running operations frees up resources on the server and
+  hints that there may be underlying performance issues (such as a lack of appropriate indexes).
+
+maxPoolSize Recommendations
++++++++++++++++++++++++++++
+
+``maxPoolSize`` specifies the maximum number of connections available in each connection pool.
+Begin with a lower value to reduce the maximum pressure each server may
+experience from each connected client application. Fewer overall connections
+in the pool means the database server is using less resources maintaining connections.
+
+
+Specify ``maxPoolSize`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxSize() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxSize(int)>`__ method.
+
+maxConnecting Recommendations
++++++++++++++++++++++++++++++
+
+``maxConnecting`` specifies the maximum number of connections to each server that
+the driver will concurrently establish before making them available in
+the connection pool.
+
+.. important:: Start with Lower Values
+
+   Start with lower values for ``maxConnecting``. This helps to avoid bursts
+   in connections in response to a temporary increase in traffic.
+
+
+Do not confuse ``maxConnecting`` with ``maxPoolSize``, which specifies the
+maximum number of connections the driver can maintain in the connection pool.
+
+.. example::
+
+   Consider a failover scenario in a three member replica set with applications under
+   heavy load. Without ``maxConnecting``, ``maxPoolSize * application servers``
+   connections would be established. With default settings for ``maxPoolSize``
+   and three application servers, that would be 300 new connections, concurrently.
+
+   With ``maxConnecting`` set to "3", the new primary will receive
+   ``maxConnecting * application servers`` connections concurrently, or nine.
+
+Specify ``maxConnecting`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxConnecting() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxConnecting(int)>`__ method.
+
+
+maxTime()
++++++++++
+
+
+``maxTime()`` specifies the *cumulative* time limit for processing operations
+in MongoDB. If an operation exceeds this limit, MongoDB terminates the
+operation.
+
+.. tip::
+
+   Use ``maxTime()`` to limit operation execution time. Do not use ``socketTimeoutMS``.
+
+   Closing a socket forces a reconnect, which slows all operations, and does not have any
+   impact on the operation that the developer is attempting to abort. MongoDB will continue
```

Comment:
Is this the "developer" or "application"? I think the developer can specify a setting, but the application performs the action.